### ISSUE TITLE
mIOU decreasing as the Lovasz Hinge Loss decreases

### ISSUE THREAD

**Issue by m-hamza-mughal** [bermanmaxim/LovaszSoftmax#21]
Hi Maxim Berman! Great Work.
I am using the lovasz_hinge loss and iou_binary as my metric. 
My labels are binary masks of foreground represented as 1 and background as 0. 
And I am currently overfitting one example just to see how my model(which is a form of hyper-network) works.
But as the Lovasz Hinge loss decreases, the output of iou_binary also decreases.
Thanks alot in advance for helping

**Comment by bermanmaxim** [comment]
Hello,
Because Lovász-hinge/softmax are relaxations it is unfortunately not always the case that decreasing them is directly correlated with increasing IoU. You could test if Lovász-sigmoid (2D version of Lovász-Softmax) performs better than Lovász-Hinge for your application.
